src/index.ts

```diff
@@ -249,9 +295,25 @@ export function HTPipe(...objs: any[]) {
         : isHasInitPreContext(right)
         ? { initPreContext: left.initPreContext }
         : {}) as PipedPreContext<any, any>),
+      ...((isHasSanitizeParams(left) && isHasSanitizeParams(right)
+        ? {
+            sanitizeParams: (context: any) => {
+              const leftOut = left.sanitizeParams(context) || {};
+              const rightIn = {
+                ...(leftOut as {}),
+              };
+              const rightOut = right.sanitizeParams(rightIn) || {};
+              return rightOut as {};
+            },
+          }
+        : isHasSanitizeParams(left)
+        ? { sanitizeParams: left.sanitizeParams }
+        : isHasSanitizeParams(right)
+        ? { sanitizeParams: right.sanitizeParams }
+        : {}) as PipedSanitizeParams<any, any>),
       ...((isHasAttachData(left) && isHasAttachData(right)
         ? {
```

Comment:
the fact that this compiled before is frightening ... we need to get to the bottom of WHY, and make it so that it wouldn't have compiled without it
- [ ] make these errors be errors

Reply:
this was compiled without error because of we use import 'mocha' in test file
when we doing this typescript take types of mocha
in node_modules/@types/mocha/types.d.ts we can find context variable
that's why it was compiled without error before